airflow/docker.go

```diff
@@ -458,6 +501,488 @@ func (d *DockerCompose) Pytest(pytestFile, customImageName, deployImageName, pyt
 	return exitCode, errors.New("something went wrong while Pytesting your DAGs")
 }
 
+func (d *DockerCompose) UpgradeTest(newAirflowVersion, deploymentID, newImageName string, conflictTest, versionTest, dagTest bool, client astro.Client) error {
+	// figure out which tests to run
+	if !conflictTest && !versionTest && !dagTest {
+		conflictTest = true
+		versionTest = true
+		dagTest = true
+	}
+	// if user supplies deployment id pull down current image
+	var deploymentImage string
+	if deploymentID != "" {
+		c, err := config.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		domain := c.Domain
+		if domain == "" {
+			return errors.New("no domain set, re-authenticate")
+		}
+		ws := c.Workspace
+		registry := GetRegistryURL(domain)
+		repository := registry + "/" + c.Organization + "/" + deploymentID
+		currentDeployment, err := deployment.GetDeployment(ws, deploymentID, "", true, client, nil)
+		if err != nil {
+			return err
+		}
+		currentImageTag := currentDeployment.DeploymentSpec.Image.Tag
+		deploymentImage = fmt.Sprintf("%s:%s", repository, currentImageTag)
+		token := c.Token
+		// Splitting out the Bearer part from the token
+		splittedToken := strings.Split(token, " ")[1]
+		fmt.Printf("\nPulling image from Astro Deployment %s\n\n", currentDeployment.Label)
+		err = d.imageHandler.Pull(registry, registryUsername, splittedToken, deploymentImage)
+		if err != nil {
+			return err
+		}
+	} else {
+		// build image for current Airflow version to get current Airflow version
+		fmt.Println("\nBuilding image for current Airflow version")
+		imageBuildErr := d.imageHandler.Build(d.dockerfile, airflowTypes.ImageBuildConfig{Path: d.airflowHome, Output: true})
+		if imageBuildErr != nil {
+			return imageBuildErr
+		}
+	}
+	// get current Airflow version
+	currentAirflowVersion, err := d.imageHandler.GetLabel(deploymentImage, RuntimeImageLabel)
+	if err != nil {
+		return err
+	}
+	if currentAirflowVersion == "" {
+		currentAirflowVersion, err = d.imageHandler.GetLabel(deploymentImage, AirflowImageLabel)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Did we forget to update code here? Why are we fetching currentAirflowVersion twice?

Reply:
if using the runtime image tag doesn't work it attempts to use the certified tag